src/NuGet.Core/NuGet.Commands/RestoreCommand/DependencyGraphResolver.cs

```diff
@@ -517,6 +517,13 @@ public DependencyGraphResolver(RestoreCollectorLogger logger, RestoreRequest res
                             libraryRangeInterningTable);
 
                         findLibraryEntryCache.Add(libraryRangeOfCurrentRef, refItemResult);
+
+                        // If the package came from a remote library provider, it needs to be installed locally.
+                        var isRemote = context.RemoteLibraryProviders.Contains(refItem.Data.Match.Provider);
+                        if (isRemote)
+                        {
+                            newRTG.Install.Add(refItem.Data.Match);
+                        }
```

Comment:
```suggestion
                        if (context.RemoteLibraryProviders.Contains(refItem.Data.Match.Provider))
                        {
                            newRTG.Install.Add(refItem.Data.Match);
                        }
```

Reply:
Will consider this as part of the refactoring